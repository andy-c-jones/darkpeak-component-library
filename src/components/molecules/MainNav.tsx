import { NavItem } from '../atoms/NavItem';

type MainNavProps = {
  /*
   * The navigation items
   */
  navItems: {
    /*
     * The label of the navigation item
     */
    label: string;
    /*
     * The href of the navigation item
     */
    href: string;
  }[];
};

/**
 * The main navigation component
 */
export const MainNav = ({ navItems }: MainNavProps) => {
  return (
    <nav className="flex gap-6 items-center flex-wrap">
      <ul className="flex gap-10 grow-0 flex-wrap">
        {navItems.map((navItem, index) => (
          <NavItem key={index} {...navItem} />
        ))}
      </ul>
    </nav>
  );
};
