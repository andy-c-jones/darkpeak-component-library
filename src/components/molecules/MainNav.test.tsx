/**
 * @vitest-environment jsdom
 */
import { render, cleanup } from '@testing-library/react';
import { describe, it, expect, beforeEach } from 'vitest';
import { MainNav } from './MainNav';

beforeEach(() => {
  cleanup();
});

describe('MainNav', () => {
  it('renders without error', () => {
    const { asFragment } = render(<MainNav navItems={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the navigation items', () => {
    const c = render(
      <MainNav
        navItems={[
          { label: 'Home', href: '/' },
          { label: 'About', href: '/about' }
        ]}
      />
    );

    let navItems = c.container.querySelectorAll('li');

    expect(navItems.length).toBe(2);

    expect(navItems[0].querySelector('a')?.textContent).toBe('Home');
    expect(navItems[0].querySelector('a')?.getAttribute('href')).toBe('/');
    expect(navItems[1].querySelector('a')?.textContent).toBe('About');
    expect(navItems[1].querySelector('a')?.getAttribute('href')).toBe('/about');
  });
});
