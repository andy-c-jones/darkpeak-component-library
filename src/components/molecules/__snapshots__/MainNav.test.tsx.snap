// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainNav > renders without error 1`] = `
<DocumentFragment>
  <nav
    class="flex gap-6 items-center flex-wrap"
  >
    <ul
      class="flex gap-10 grow-0 flex-wrap"
    />
  </nav>
</DocumentFragment>
`;
